### PR TITLE
move plone.allowdiscussion behavior to plone.app.discussion

### PR DESCRIPTION
**Pull request by jensens**
see https://github.com/plone/plone.app.discussion/pull/211

### CODE REVIEW

File: plone/app/dexterity/behaviors/discussion.py [hunk 1]
```diff
@@ -1,39 +1,6 @@
...
+from zope.deferredimport import deprecated
 
-
-options = SimpleVocabulary(
-    [
-        SimpleTerm(value=True, title=_("Yes")),
-        SimpleTerm(value=False, title=_("No")),
-    ]
+deprecated(
+    "IAllowDiscussion is here deprecated. Import from plone.app.z3cform.widgets.select instead (will be removed in Plone 7)"
```
Comment (davisagli) [inline plone/app/dexterity/behaviors/discussion.py:4]:
```suggestion
    "IAllowDiscussion is here deprecated. Import from plone.app.discussion.behaviors instead (will be removed in Plone 7)"
```

File: plone/app/dexterity/behaviors/discussion.py [hunk 2]
```diff
@@ -1,39 +1,6 @@
...
+from zope.deferredimport import deprecated
 
-
-options = SimpleVocabulary(
-    [
-        SimpleTerm(value=True, title=_("Yes")),
-        SimpleTerm(value=False, title=_("No")),
-    ]
+deprecated(
+    "IAllowDiscussion is deprecated here. Import from plone.app.discussion.behaviors instead (will be removed in Plone 7)"
```
Comment (gforcada) [inline plone/app/dexterity/behaviors/discussion.py:4]: Do we want to say `deprecated` at all? 🤔 

If I see _deprecated_ somewhere I get that the functionality is being __removed__.

Here rather we want to say that it has been __moved__ the two letters' diff makes a difference 😆 

Something like: `Please adjust your imports, IAllowDiscussion is now in p.a.discussion`

**Review by davisagli — APPROVED** [review]
LGTM once the other parts of this are ready

**Review by mauritsvanrees — APPROVED** [review]
LGTM.

I have removed `plone.app.discussion` from the test extras.

**Review by gforcada — APPROVED** [review]

**Pull request merged**